https://bugzilla.gnome.org/show_bug.cgi?id=675743
http://git.gnome.org/browse/glib/commit/?id=431d3adacdd640141f83a6c3496760adc6aba5eb

system libelf is not sufficient for glib (lacks elf_getshdrstrndx and
elf_getshdrnum) so let's make sure we don't define HAVE_LIBELF

--- glib-2.32.3/configure.ac
+++ glib-2.32.3/configure.ac
@@ -1822,10 +1822,10 @@
 
 if test $have_libelf = maybe; then
   glib_save_LIBS=$LIBS
-  AC_CHECK_LIB([elf], [elf_begin], , [have_libelf=no])
-  AC_CHECK_LIB([elf], [elf_getshdrstrndx], , [have_libelf=no])
-  AC_CHECK_LIB([elf], [elf_getshdrnum], , [have_libelf=no])
-  AC_CHECK_HEADER([libelf.h], , [have_libelf=no])
+  AC_CHECK_LIB([elf], [elf_begin], [:], [have_libelf=no])
+  AC_CHECK_LIB([elf], [elf_getshdrstrndx], [:], [have_libelf=no])
+  AC_CHECK_LIB([elf], [elf_getshdrnum], [:], [have_libelf=no])
+  AC_CHECK_HEADER([libelf.h], [:], [have_libelf=no])
   LIBS=$glib_save_LIBS
 
   if test $have_libelf != no; then
